app/src/main/java/com/example/goodnote/database/LocalDb.kt

```diff
@@ -42,21 +40,68 @@ abstract class LocalDb : RoomDatabase() {
     }
 
     private class LocalDbCallback : RoomDatabase.Callback() {
+
         override fun onCreate(db: SupportSQLiteDatabase) {
             super.onCreate(db)
+
             instance?.let { database ->
                 GlobalScope.launch {
```

Comment:
We have one issue here. On the first start of the app, the database won't be initialized when you try to fill it with data. 

In order to prefill data on the first app start, you should move this code in the `buildDb` method, and call this logic after room object has been initialized. 

You can also use the second option, where you use `SupportSQLiteDatabase` db parameter of the `onCreate` method and apply raw SQLite commands via this object to insert data to db. 

IMHO,  first option is much easier. :)

Reply:
It weirdly didn't populate with
 `return Room.databaseBuilder(context, LocalDb::class.java, DB_NAME).build().also { populate(it) }`
but did once the `also` wasn't used.

Reply:
` private fun buildDb(context: Context): LocalDb {

            val db = Room.databaseBuilder(context, LocalDb::class.java, DB_NAME).addCallback(LocalDbCallback()).build()

populate(db)

return db
        }`

This sometimes populates db, sometimes loads 7/9 notes, sometimes loads tags, sometimes it does not. Reverted back to using `RoomDatabase.Callback()`